runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -894,15 +894,22 @@ Java_java_lang_Class_getMethodImpl(JNIEnv *env, jobject recv, jobject name, jobj
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
 	j9object_t resultObject = NULL;
 	vmFuncs->internalEnterVMFromJNI(currentThread);
-	J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
 
 	PORT_ACCESS_FROM_VMC(currentThread);
 	if ((NULL == name) || (NULL == partialSignature)) {
 		vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
 	} else {
-		J9ROMClass *romClass = clazz->romClass;
-		/* primitives/arrays don't have local methods */
-		if (!J9ROMCLASS_IS_PRIMITIVE_OR_ARRAY(romClass)) {
+		J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
+		J9ROMClass *romClass = NULL;
+		
+		if (J9CLASS_IS_ARRAY(clazz)) {
+			/* Use underlying type for arrays */
+			romClass = ((J9ArrayClass*)clazz)->leafComponentType->romClass;
```

Comment:
I read the javadoc as allowing array classes to find methods in Object, not methods in the component type. Do you have any test cases?

Also, how does this restrict clone()?

Reply:
I see, you are modifying the !J9ROMCLASS_IS_PRIMITIVE_TYPE test. This also has side effects because the romClass is used to set the lookupFLags. Just change the !J9ROMCLASS_IS_PRIMITIVE_TYPE test directory rather than modifying the romClass.

Reply:
> I read the javadoc as allowing array classes to find methods in Object, not methods in the component type. Do you have any test cases?

Manually verified that `RI` has following behaviours:
`System.out.println(new TestClassReflect[2].getClass().getMethod("getClass")`
returns 
`class [Latestse.java_lang_class.TestClassReflect;.getMethod(getClass) = public final native java.lang.Class java.lang.Object.getClass()`,
and 
`System.out.println(new TestClassReflect[2].getClass().getMethod("clone")` 
throws 
`java.lang.NoSuchMethodException: [Latestse.java_lang_class.TestClassReflect;.clone()`
and 
`System.out.println(new TestClassReflect[2].getClass().getMethod("method0")` 
throws 
`java.lang.NoSuchMethodException: [Latestse.java_lang_class.TestClassReflect;.method0()`
in which `method0` is a public method within class `TestClassReflect`.
`OpenJ9` with this PR matches the behaviours above.
Note: these behaviours are observed in `RI JDK8` as well.      

> Also, how does this restrict clone()?

`java.lang.Class.getMethod()` only find `public` method while `clone` is a `protected` method.

Reply:
> Just change the !J9ROMCLASS_IS_PRIMITIVE_TYPE test directly rather than modifying the romClass.

Agreed, PR has been updated. Also verified that `OpenJ9` with this change matches `RI` behaviours in question. Please have another look.